python-sdk/src/astro/databases/mysql.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pandas as pd
+import sqlalchemy
+from airflow.providers.mysql.hooks.mysql import MySqlHook
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from astro.databases.base import BaseDatabase
+from astro.options import LoadOptions
+from astro.settings import MYSQL_SCHEMA
+from astro.table import BaseTable, Metadata
+from astro.utils.compat.functools import cached_property
+
+DEFAULT_CONN_ID = MySqlHook.default_conn_name
+if TYPE_CHECKING:  # pragma: no cover
+    pass
+
+
+class MysqlDatabase(BaseDatabase):
+    DEFAULT_SCHEMA = MYSQL_SCHEMA
+
+    _create_schema_statement: str = (
+        "CREATE SCHEMA IF NOT EXISTS {} "
+        "DEFAULT CHARACTER SET = 'utf8' DEFAULT COLLATE = "
+        "'utf8_unicode_ci'"
+    )
+
+    def __init__(
+        self,
+        conn_id: str = DEFAULT_CONN_ID,
+        table: BaseTable | None = None,
+        load_options: LoadOptions | None = None,
+    ):
+        super().__init__(conn_id)
+        self.table = table
+        self.load_options = load_options
+
+    @cached_property
+    def hook(self) -> MySqlHook:
+        """Retrieve Airflow hook to interface with the mysql database."""
+        conn = MySqlHook(mysql_conn_id=self.conn_id).get_connection(self.conn_id)
+        kwargs = {}
+        if conn.schema is None:
+            if (
+                self.table
+                and self.table.metadata
+                and (self.table.metadata.database or self.table.metadata.schema)
+            ):
+                kwargs.update({"schema": self.table.metadata.database or self.table.metadata.schema})
```

Comment:
Should we throw an error if a user has both the metadata schema and database?

Reply:
made the change